src/functions/TestResults.JUnit4.ps1

```diff
@@ -42,23 +42,16 @@ function Write-JUnitTestSuiteElements {
 
     $XmlWriter.WriteStartElement('testsuite')
 
-    if ('File' -eq $Container.Type) {
-        $path = $Container.Item.FullName
+    if ($container.Type -notin 'File', 'ScriptBlock') {
+        throw "Container type '$($container.Type)' is not supported."
```

Comment:
Maybe limit this in constructors or setters instead of validating everywhere?

Reply:
Yes please.